### DISCUSSION TITLE
Using CPU instead of GPU

### DISCUSSION THREAD

**Discussion by Huba2004** [coqui-ai/TTS#2208] [Archived]
.

**Comment by erogol** [comment]
Nope, you need a GPU unfortunately.

**Comment by thefex** [comment]
There is a possibility to check if MPS (Apple Metal -> M1 GPU support) via torch.backends.is_mps_available() and convert .cuda() calls into .to(torch.device('mps')) however...
I tried and failed - this is a blocker: https://github.com/pytorch/pytorch/issues/88415 - if that gets resolved that *should* be ~1 hour job.

Generally speaking it should be doable anyway ( but I am not python dev/experienced ML dev at all so it would take me too much time ) by replacing to specific MPS engine commands. Unfortunately MPS engine is not always as friendly as cuda - in CUDA pytorch simple .cuda() calls works.

You also need to make sure to update requirements.txt numba=0.55.1 to 0.55.2 as Apple M1 is supported in 0.55.2 (without that compile fails -> https://github.com/numba/numba/issues/7951)

If someone experienced could pick this.. M1 machines/gpus are very strong and can yield better results that TESLA T4 etc..

**Reply by ROBERT-MCDOWELL** [reply]
my advice, learn python and DIY
